mule-extensions-api/src/main/java/org/mule/runtime/extension/api/options/ConfigurationAware.java

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.options;
+
+public interface ConfigurationAware<T> {
```

Comment:
not sure what you use this for, but doesn't soundl ike something we want to have